kyaml/yaml/kfns.go

```diff
@@ -40,6 +44,70 @@ func ClearEmptyAnnotations(rn *RNode) error {
 	return nil
 }
 
+// k8sDataSetter place key value pairs in either a 'data' or 'binaryData' field.
+// Useful for creating ConfigMaps and Secrets.
+type k8sDataSetter struct {
+	Key             string `yaml:"key,omitempty"`
+	Value           string `yaml:"value,omitempty"`
+	ProtectExisting bool   `yaml:"protectExisting,omitempty"`
+}
+
+func (s k8sDataSetter) Filter(rn *RNode) (*RNode, error) {
+	if !utf8.Valid([]byte(s.Value)) {
+		// Core k8s ConfigMaps store k,v pairs with 'v' passing the above utf8
+		// test in a mapping field called "data" as a string. Pairs with a 'v'
+		// failing this test go into a field called binaryData as a []byte.
+		// TODO: support this distinction in kyaml with NodeTagBytes?
+		return nil, fmt.Errorf(
+			"key '%s' appears to have non-utf8 data; "+
+				"binaryData field not yet supported", s.Key)
+	}
+	keyNode, err := rn.Pipe(Lookup(DataField, s.Key))
+	if err != nil {
+		return nil, err
+	}
+	if keyNode != nil && s.ProtectExisting {
+		return nil, fmt.Errorf(
+			"protecting existing %s='%s' against attempt to add new value '%s'",
+			s.Key, strings.TrimSpace(keyNode.MustString()), s.Value)
+	}
+	v := NewScalarRNode(s.Value)
+	v.YNode().Tag = NodeTagString
+	// Add quotes?
+	// v.YNode().Style = yaml.SingleQuotedStyle
```

Comment:
Probably don't need this -- but you may want to use `FormatNonStringStyle()` to make sure it gets quoted iff and only if the contents parse as non-string values -- e.g. `true`

Reply:
done (added todo since schema not available at this point yet)